src/web/page/web_page.rs

```diff
@@ -115,51 +95,47 @@ macro_rules! impl_axum_webpage {
 /// the context.
 #[derive(Clone)]
 pub(crate) struct DelayedTemplateRender {
-    pub template: String,
-    pub context: Context,
+    pub template: Arc<Box<dyn AddCspNonce + Send + Sync>>,
+    #[allow(dead_code)]
     pub cpu_intensive_rendering: bool,
 }
 
 fn render_response(
     mut response: AxumResponse,
-    templates: Arc<TemplateData>,
     csp_nonce: String,
 ) -> BoxFuture<'static, AxumResponse> {
     async move {
         if let Some(render) = response.extensions_mut().remove::<DelayedTemplateRender>() {
-            let DelayedTemplateRender {
-                template,
-                mut context,
-                cpu_intensive_rendering,
-            } = render;
-            context.insert("csp_nonce", &csp_nonce);
-
-            let rendered = if cpu_intensive_rendering {
-                templates
-                    .render_in_threadpool(move |templates| {
-                        templates
-                            .templates
-                            .render(&template, &context)
-                            .map_err(Into::into)
-                    })
-                    .await
-            } else {
-                templates
-                    .templates
-                    .render(&template, &context)
-                    .map_err(Error::new)
-            };
-
-            let rendered = match rendered {
+            let DelayedTemplateRender { template, .. } = render;
+
+            // let rendered = if cpu_intensive_rendering {
```

Comment:
I'm not sure if it's still relevant to have this code. I commented it to ensure it was ok.

Reply:
Sounds goo to me! Then moving this content in there.